### DISCUSSION TITLE
Quantity of running workers and number of worker child processes in k8s environment.

### DISCUSSION THREAD

**Discussion by quizmoon** [taskiq-python/taskiq#193] [General]
As example, in case of FastAPI it is recommended to quit from using gunicorn to run multiple processes in image and use just uvicorn with one processes, but using replicaset.

What is preferable for taskiq?
a) deploy one worker image with XX child processes
b) deploy XX workers images using replicaset with just one child processes in each.

**Comment by s3rius** [comment]
Taskiq doesn't have any preferences in how you run it. So, it's up to you to decide. 

However, deploying multiple pods, using replicaset might be a better solution, because it allows you to build HPAs that scale workers, depending on the number of messages in queue or based on some other related metric.

**Reply by quizmoon** [reply]
In case of using replicaset, how many child processes would you recommend to run per worker image?

**Reply by s3rius** [reply]
I would recommend using 2 worker processes. It's a bit faster than having one and not that memory heavy. That's why I defined 2 worker processes as default value.
